src/types/toasts.types.ts

```diff
@@ -29,17 +30,32 @@ export interface MessageIconToastType extends SharedIconToastType {
    */
   status?: string;
   transaction?: never;
+  component?: never;
 }
 
 export interface TransactionIconToastType extends SharedIconToastType {
   transaction: ServerTransactionType;
+  component?: never;
```

Comment:
If you don't know the type is better to use `unknown` type. For `component` I think you should know the type